Replicator/tests/ReplicatorCollectionSGTest.cc

```diff
@@ -2244,3 +2245,202 @@ TEST_CASE_METHOD(ReplicatorCollectionSGTest, "Give SG a rev history with a gap",
         REQUIRE(revID.hasPrefix(to_string(numRevs1 + numInitialRevs) + "-"));
     }
 }
+
+// To run this test, the sync function for Tulips collection must be,
+// "tulips":{"sync":"function(doc,olddoc){if(doc.cbl_445==\"bad\")throw({\"forbidden\":\"read_only\"});channel(doc.channels)}"}
```

Comment:
Maybe create a new collection with this sync function? So SG config doesn't need to be updated before running this test.

Reply:
This sync function wouldn't hurt other tests. I used persistent config. I don't know if I can add a new collection after collections are registered with PUT a new db.

Reply:
You can add another collection in the PUT request.

Reply:
{"error":"Precondition Failed","reason":"Duplicate database name \"scratch\""}